packages/frontend-2/pages/projects/[id]/webhooks/index.vue

```diff
@@ -0,0 +1,269 @@
+          {{
+            (item.triggers as string[])
+              .map(
+                (event, index, array) =>
+                  `"${event}"${index < array.length - 1 ? ',' : ''}`
+              )
+              .join(' ')
+          }}
+        </div>
+      </template>
+    </LayoutTable>
+
+    <ProjectWebhooksPageDeleteWebhookDialog
+      v-model:open="showDeleteWebhookDialog"
+      :webhook="webhookToModify"
+    />
+
+    <ProjectWebhooksPageEditWebhookDialog
+      v-model:open="showEditWebhookDialog"
+      :webhook="webhookToModify"
+    />
+
+    <ProjectWebhooksPageCreateWebhookDialog
+      v-model:open="showNewWebhookDialog"
+      :stream-id="projectId"
+      @webhook-created="handleWebhookCreated"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue'
+import { useMutation, useQuery } from '@vue/apollo-composable'
+import {
+  PlusIcon,
+  BookOpenIcon,
+  InformationCircleIcon,
+  CheckCircleIcon,
+  XCircleIcon
+} from '@heroicons/vue/20/solid'
+import { TrashIcon, PencilIcon } from '@heroicons/vue/24/outline'
+import {
+  FormSwitch,
+  ToastNotificationType,
+  TableItemType
+} from '@speckle/ui-components'
+import { projectWebhooksQuery } from '~~/lib/projects/graphql/queries'
+import { updateWebhookMutation } from '~~/lib/projects/graphql/mutations'
+import { useGlobalToast } from '~~/lib/common/composables/toast'
+import { projectRoute } from '~~/lib/common/helpers/route'
+import { isWebhook } from '~~/lib/projects/helpers/utils'
+import { WebhookItem } from '~~/lib/projects/helpers/types'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+
+definePageMeta({
+  middleware: ['require-valid-project']
+})
+
+const { triggerNotification } = useGlobalToast()
+const { mutate: updateMutation } = useMutation(updateWebhookMutation)
+const route = useRoute()
+
+const webhookToModify = ref<TableItemType<WebhookItem> | null>(null)
+const showDeleteWebhookDialog = ref(false)
+const showEditWebhookDialog = ref(false)
+const showNewWebhookDialog = ref(false)
+
+const projectId = computed(() => route.params.id as string)
+const projectName = computed(() => pageResult.value?.project?.name || 'Unknown Project')
+const webhooks = computed<WebhookItem[]>(() => {
+  return (
+    pageResult.value?.project?.webhooks?.items?.map(
+      (webhook) => webhook as WebhookItem
+    ) || []
+  )
+})
+
+const getHistoryStatus = (item: TableItemType<WebhookItem>) => {
+  const recentHistory = item.history?.items?.[0]
+  if (recentHistory) {
+    switch (recentHistory.status) {
+      case 1:
+        return 'alert'
+      case 2:
+        return 'called'
+      case 3:
+        return 'error'
+    }
+  } else {
+    return 'noEvents'
+  }
+}
+
+const getHistoryStatusInfo = (item: TableItemType<WebhookItem>) => {
+  if (isWebhook(item)) {
+    const recentHistory = item.history?.items?.[0]
+    if (recentHistory) {
```

Comment:
This can be shortened to a ternary condition: `return recentHistory ? recentHistory.statusInfo : 'No events yet'`